Support restarting Nomad without restarting nspawn containers

It seems that restarting Nomad service (for example when upgrading Nomad or reloading configuration) restarts jobs run by nspawn driver. Docker jobs stay alive and are not restarted when restarting Nomad.

I observed the following errors in logs:
>   2020-10-26T19:28:26.592+0100 [ERROR] client.alloc_runner.task_runner: error recovering task; cleaning up: alloc_id=2fb194a7-5964-07f6-e9da-b3c09abfb3a5 task=<my nspawn task> error="rpc error: code = Unknown desc = failed to decode driver config: EOF" task_id=2fb194a7-5964-07f6-e9da-b3c09abfb3a5/<my nspawn task>/0af76d99
    2020-10-26T19:28:26.592+0100 [WARN]  client.alloc_runner.task_runner: error destroying unrecoverable task: alloc_id=2fb194a7-5964-07f6-e9da-b3c09abfb3a5 task=<my nspawn task> error="rpc error: code = Unknown desc = task not found for given id" task_id=2fb194a7-5964-07f6-e9da-b3c09abfb3a5/<my nspawn task>/0af76d99

Failed jobs are then reallocated and run fine, however, it's undesirable that they are restarted. 
Would it be hard to support that?

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [Unreleased]
 ### Fixed
   - Ensure all test containers are properly stopped.
+  - Fixed a bug which caused tasks to not be recovered on Nomad restarts (#17)
 
 ## [0.4.0] - 2020-10-03
 ### Added

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,10 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [Unreleased]
+### Fixed
+  - Ensure all test containers are properly stopped.
+
 ## [0.4.0] - 2020-10-03
 ### Added
   - Support for granting additional capabilities to containers. @mateuszlewko
@@ -27,6 +31,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
     logs and shown in the WebUi.
   - Support for `volume_mount` stanza. Volumes are enabled by default and can be
     disabled by setting `volumes = false` inside the driver config.
+
 ## [0.2.0] - 2020-06-27
 ### Added
   - Support for downloading images via the

--- a/nspawn/driver.go
+++ b/nspawn/driver.go
@@ -258,11 +258,6 @@ func (d *Driver) RecoverTask(handle *drivers.TaskHandle) error {
 		return fmt.Errorf("failed to decode task state from handle: %v", err)
 	}
 
-	var driverConfig drivers.TaskConfig
-	if err := handle.Config.DecodeDriverConfig(&driverConfig); err != nil {
-		return fmt.Errorf("failed to decode driver config: %v", err)
-	}
-
 	plugRC, err := pstructs.ReattachConfigToGoPlugin(taskState.ReattachConfig)
 	if err != nil {
 		return fmt.Errorf("failed to build ReattachConfig from taskConfig state: %v", err)

--- a/nspawn/driver_test.go
+++ b/nspawn/driver_test.go
@@ -362,5 +362,6 @@ func TestNspawnDriver_HandlerExec(t *testing.T) {
 		t.Fatalf("expected output to contain %q but found: %q", expected, res.Stdout)
 	}
 
+	require.NoError(harness.StopTask(task.ID, 10 * time.Second, ""))
 	require.NoError(harness.DestroyTask(task.ID, true))
 }

